### PR TITLE
Add dist-rmse score, addresses #5

### PR DESCRIPTION
**Pull request by Cecca**
Compute the root mean squared error between the true k-nn distances and
the reported k-nn distances.

The idea is that a low value means that we are not including in the
reported points a point that is too far away.

### CODE REVIEW

File: data_export.py [hunk 1]
```diff
@@ -135,8 +136,8 @@
         dataset = get_dataset(dataset_name)
         unique_algorithms = get_unique_algorithms()
         results = load_all_results(dataset_name, count, True, args.batch)
+        print(results)
```
Comment (maumueller) [inline data_export.py:139]: Do you really want to keep this in here?

**Comment by maumueller** [comment]
I totally forgot that we also have "relative distances" already in ann-benchmarks. It's value is the sum of distances of the true NN and the reported NN.

glove-rmse

glove-rel

I think the relative distance also has the advantage that it provides some normalization to the distances that are usually found in the dataset, whereas the dist-rmse score doesn't as far as I can see. (E.g., for mnist we usually have distances in the thousands.)

File: ann_benchmarks/plotting/metrics.py [hunk 2]
```diff
@@ -28,6 +28,19 @@ def knn(dataset_distances, run_distances, count, metrics, epsilon=1e-10):
...
+        rmse_metrics = metrics.create_group('dist-rmse')
+        top_true = dataset_distances[:,0:count]
+        diff = run_distances - top_true
+        rmse = np.sqrt(np.sum(diff*diff, axis=1))
+        rmse_metrics.attrs['mean'] = np.mean(rmse)
+        rmse_metrics.attrs['std'] = np.std(rmse)
+    else:
+        pass
+    return metrics['dist-rmse']
+
```
[the remark this answers is not in the document]
Comment (Cecca) [reply ann_benchmarks/plotting/metrics.py:44]: Thank you!

**Comment by Cecca** [comment]
Very nice! Then I think we can just use the relative distance, though they seem to have a very similar behaviour.

Shall we just close this PR?

**Pull request closed without merging**